src/librustc_borrowck/diagnostics.rs

```diff
@@ -286,6 +286,54 @@ You can read more about cell types in the API documentation:
 https://doc.rust-lang.org/std/cell/
 "##,
 
+E0389: r##"
+An attempt was made to assign to data in a non-mutable reference. Example:
```

Comment:
Sure, but before I'd prefer this RFC to be definitely accepted. Once done, you can do it if you want (it'd be very appreciated actually!).